src/v2/Components/ArtworkFilter/SavedSearch/Components/FiltersPills.tsx

```diff
@@ -20,17 +36,36 @@ export const FiltersPills: FC<FiltersPillsProps> = ({
   pills,
   savedSearchAttributes,
 }) => {
+  const filterContext = useArtworkFilterContext()
+
+  const removePill = (pill: FilterPill) => {
+    if (pill.isDefault) return
+
+    let filter = filterContext?.currentlySelectedFilters?.()[pill.filterName]
+
+    if (isArray(filter)) {
+      filter = filter.filter(value => value !== pill.name)
+      filterContext.setFilter(pill.filterName as keyof ArtworkFilters, filter)
+    } else {
+      filterContext.setFilter(
+        pill.filterName as keyof ArtworkFilters,
+        initialArtworkFilterState[pill.filterName]
+      )
+    }
+  }
+
   return (
     <>
       <Flex flexWrap="wrap" mx={-PILL_HORIZONTAL_MARGIN_SIZE}>
-        {pills.map((pill, index) => (
+        {pills.map(pill => (
           <Pill
-            key={`filter-label-${index}`}
+            key={`filter-label-${pill.name}`}
             variant="textSquare"
             mx={PILL_HORIZONTAL_MARGIN_SIZE}
             mb={1}
+            onClick={() => removePill(pill)}
```

Comment:
**question**: Might be worth clarifying with design whether the whole pill should be clickable - and trigger the removal of that filter criteria - or if only the close icon X should be clickable for the removal. It's a bit unclear from the Figma specs what's preferred.

Reply:
@dblandin it's been answered [here](https://artsyproduct.atlassian.net/browse/FX-3543#:~:text=Should%20tapping%20anywhere,and%20filter%20value), so a criteria should be removed on pill tap, not icon

Reply:
Ahh, I see that Q&A now:

> Should tapping anywhere on the pill dismiss, or only on the X?
> - Tapping a pill with a 'X' should remove the pill and filter value
> - Artist pill should not have an X

Thanks!